### PR TITLE
implemented all required tests

### PR DESCRIPTION
**Pull request by 3xactCoder**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/impl/FileReaderImpl.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't properly read file from path: " + filePath, e);
+        }
+        return lines;
+    }
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/service/impl/FileReaderImpl.java:26]: remove redundant empty lines

File: src/main/java/core/basesyntax/service/impl/FileWriterImpl.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    @Override
+    public void write(String report, String filePath) {
+        try (BufferedWriter writer = new BufferedWriter(new java.io.FileWriter(filePath))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write report to file: " + filePath, e);
+        }
+    }
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/service/impl/FileWriterImpl.java:17]: remove redundant empty lines

File: src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            if (handler == null) {
+                throw new RuntimeException("No handler found for operation: "
+                        + transaction.getOperation());
+            }
+            handler.handle(transaction);
+        }
+    }
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java:28]: remove redundant empty lines

check all files

File: src/main/java/core/basesyntax/Main.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import core.basesyntax.strategy.ReturnOperation;
+import core.basesyntax.strategy.SupplyOperation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) {
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.read("reportToRead.csv");
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Main.java]: file should be stored in `main/resources` folder
create constant for file path

File: src/main/java/core/basesyntax/Main.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        ShopService shopService = new ShopServiceImpl(operationStrategy);
+        shopService.process(transactions);
+
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
+        String resultingReport = reportGenerator.getReport();
+
+        FileWriter fileWriter = new FileWriterImpl();
+        fileWriter.write(resultingReport, "finalReport.csv");
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/Main.java]: same comment here

File: src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Test
+    public void convertToTransaction_validLines_ok() {
+        List<String> inputLines = List.of(
+                "type,fruit,quantity",
+                "b,banana,20"
+        );
+
+        List<FruitTransaction> result = dataConverter.convertToTransaction(inputLines);
+        Assertions.assertEquals(1, result.size());
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java]: use static imports
check all places

File: src/test/java/core/basesyntax/service/impl/FileReaderImplTest.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Test
+    public void read_invalidPath_notOk() {
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            fileReader.read(INVALID_FILE_PATH);
+        });
+    }
+}
+
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/impl/FileReaderImplTest.java]: add test to read empty file

File: src/test/java/core/basesyntax/service/impl/FileReaderImplTest.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+
+
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/impl/FileReaderImplTest.java]: add test class for FileWriter

File: src/test/java/core/basesyntax/service/impl/ReportGeneratorImplTest.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    private ReportGenerator reportGenerator;
+
+    @BeforeEach
+    public void setUp() {
+        reportGenerator = new ReportGeneratorImpl();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        Storage.getAllFruits().keySet().forEach(fruit -> Storage.updateQuantity(fruit, 0));
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/impl/ReportGeneratorImplTest.java]: it's better just ot clear storage

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]
well done
